src/Acquisition/__init__.py

```diff
@@ -1,8 +1,833 @@
+from __future__ import absolute_import, print_function
+
+# pylint:disable=W0212,R0911,R0912
+
+import os
+import operator
+import sys
+import types
+
+import ExtensionClass
+
 from zope.interface import classImplements
 
-from _Acquisition import *
-from interfaces import IAcquirer
-from interfaces import IAcquisitionWrapper
+from .interfaces import IAcquirer
+from .interfaces import IAcquisitionWrapper
+
+class Acquired(object):
+    "Marker for explicit acquisition"
+
+
+_NOT_FOUND = object()  # marker
+
+###
+# Helper functions
+###
+
+
+def _has__of__(obj):
+    """Check whether an object has an __of__ method for returning itself
+    in the context of a container."""
+    # Note specifically this is a type check, not duck typing, or
+    # we get into cycles
+    return isinstance(obj, ExtensionClass.Base)
+
+
+def _apply_filter(predicate, inst, name, result, extra, orig):
+    return predicate(orig, inst, name, result, extra)
```

Comment:
Basically, because that's the way the C implementation does it. The order of arguments to the user-provided predicate function is *different* from the order of arguments passed to the helper as written in the C (and Python code) so calling the predicate directly would introduce a (gratuitous?) difference that's likely to get mistakenly changed in maintenance.  Plus, when the helper function is called, the caller's variable names are not really meaningful in terms of the arguments to the predicate function; the helper is a layer of abstraction that, to me at least,  helps smooth over the impedance mismatch.